src/vs/editor/common/config/commonEditorConfig.ts

```diff
@@ -415,10 +400,10 @@ export abstract class CommonEditorConfiguration extends Disposable implements IC
 			return;
 		}
 		const newOptions = deepCloneAndMigrateOptions(_newOptions);
-		if (CommonEditorConfiguration._subsetEquals(this._rawOptions, newOptions)) {
+		if (CommonEditorConfiguration._topLevelSubsetEquals(this._rawOptions, newOptions)) {
 			return;
 		}
-		this._rawOptions = objects.mixin(this._rawOptions, newOptions || {});
+		this._rawOptions = Object.assign(this._rawOptions, newOptions || {});
```

Comment:
This change also breaks existing semantics. Same example as above:
```
// by default minimap.showSlider is 'mouseover'
editor.updateOptions({ minimap: { enabled: true, showSlider: 'always' } });
editor.updateOptions({ minimap: { enabled: false } }); // effective value: { enabled: false, showSlider: 'always' }
```

If I understand correctly the proposed change, the effective value will now be `{ enabled: false }` instead of `{ enabled: false, showSlider: 'always' }`